stdlib/inspect.pyi

```diff
@@ -48,30 +37,53 @@ def getmembers(object: object, predicate: Optional[Callable[[Any], bool]] = ...)
 def getmodulename(path: str) -> Optional[str]: ...
 def ismodule(object: object) -> TypeGuard[ModuleType]: ...
 
-# TODO: use TypeGuard[Type[Any]] after python/mypy#10486 is resolved
-def isclass(object: object) -> bool: ...
+_T = TypeVar("_T")
+
+def isclass(object: Union[Type[_T], object]) -> TypeGuard[Type[_T]]: ...
```

Comment:
- What does `isclass("lol")` return? Does it return `TypeGuard[Type[Any]]`? I'm not sure what happens when the union picks `object` instead of `Type[_T]`, and `_T` is used only in the return type. Maybe it would be better to use explicit overloads?
- Nit: Use `Foo | Bar` instead of `Union[Foo, Bar]`

These also apply to other things.

Reply:
the function would return `False` so the type guard wouldnt be applied, i dont see how an overload would help here.

Reply:
Maybe you understand it better than I do. Then it's fine :) I just avoid using a TypeVar in only one place in general, and if the first `Type[_T]` part is ignored, we use `_T` only in the return type.

Reply:
Maybe:

```suggestion
@overload
def isclass(object: Type[_T]) -> TypeGuard[Type[_T]]: ...
@overload
def isclass(object: object) -> bool: ...
```

So when we know for sure that the given argument is a class, and we even know which class it is, we can return a `TypeGuard`. Otherwise, it may or may not be a class, and we can return `bool`.

Reply:
see this example from [pep 647](https://www.python.org/dev/peps/pep-0647/)
```py
def is_str_list(val: List[object], allow_empty: bool) -> TypeGuard[List[str]]:
    if len(val) == 0:
        return allow_empty
    return all(isinstance(x, str) for x in val)

_T = TypeVar("_T")

def is_set_of(val: Set[Any], type: Type[_T]) -> TypeGuard[Set[_T]]:
    return all(isinstance(x, type) for x in val)
```

typeguard isnt a normal a type like others, its just saying "the value passed to the function is the type passed to the typeguard's generic when this function returns `True`"

Reply:
I know what a TypeGuard is, and I think it makes sense only when we know what type should go inside `TypeGuard[...]`.

Reply:
Now that I think about it, maybe mypy will use the surrounding code to infer `_T` when the `isclass()` call alone is not enough? As I said, I don't know how exactly mypy figures this out, and your code may actually work.